test/acceptance/test/profiles_test.go

```diff
@@ -0,0 +1,171 @@
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Weave GitOps Profiles API", func() {
+	var (
+		namespace        = "test-namespace"
+		appRepoRemoteURL string
+		tip              TestInputs
+		profilesSvc      = "profiles-server"
+		clientSet        *kubernetes.Clientset
+		kClient          client.Client
+	)
+
+	BeforeEach(func() {
+		Expect(FileExists(WEGO_BIN_PATH)).To(BeTrue())
+		Expect(GITHUB_ORG).NotTo(BeEmpty())
+
+		_, _, err := ResetOrCreateCluster(namespace, true)
+		Expect(err).NotTo(HaveOccurred())
+
+		private := true
+		tip = generateTestInputs()
+		_ = initAndCreateEmptyRepo(tip.appRepoName, gitproviders.GitProviderGitHub, private, GITHUB_ORG)
+
+		clientSet, kClient = buildKubernetesClients()
+	})
+
+	AfterEach(func() {
+		deleteRepo(tip.appRepoName, gitproviders.GitProviderGitHub, GITHUB_ORG)
+	})
+
+	It("Works", func() {
+		By("Installing the Profiles API and setting up the profile helm repository")
+		appRepoRemoteURL = "git@github.com:" + GITHUB_ORG + "/" + tip.appRepoName + ".git"
+		installAndVerifyWego(namespace, appRepoRemoteURL)
+		deployProfilesHelmRepository(kClient, namespace)
+
+		By("Getting a list of profiles")
+		resp, statusCode, err := kubernetesDoRequest(namespace, profilesSvc, "8000", "/v1/profiles", clientSet)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(statusCode).To(Equal(http.StatusOK))
+
+		profiles := pb.GetProfilesResponse{}
+		Expect(json.Unmarshal(resp, &profiles)).To(Succeed())
+		Expect(profiles.Profiles).NotTo(BeNil())
+		Expect(profiles.Profiles).To(ConsistOf(&pb.Profile{
+			Name:        "podinfo",
+			Home:        "https://github.com/stefanprodan/podinfo",
+			Description: "Podinfo Helm chart for Kubernetes",
+			Sources:     []string{"https://github.com/stefanprodan/podinfo"},
+			Maintainers: []*pb.Maintainer{
+				{
+					Name:  "stefanprodan",
+					Email: "stefanprodan@users.noreply.github.com",
+				},
+			},
+			//These have to not be nil for the assertion to work. because proto :shrug:
+			Keywords:    []string{},
+			Annotations: map[string]string{},
+		}))
+
+		By("Getting the values for a profile")
+		resp, statusCode, err = kubernetesDoRequest(namespace, profilesSvc, "8000", "/v1/profiles/podinfo/6.0.1/values", clientSet)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(statusCode).To(Equal(http.StatusOK))
+
+		profileValues := pb.GetProfileValuesResponse{}
+		Expect(json.Unmarshal(resp, &profileValues)).To(Succeed())
+		values, err := base64.StdEncoding.DecodeString(profileValues.Values)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(string(values)).To(ContainSubstring("# Default values for podinfo"))
+	})
+})
+
+func buildKubernetesClients() (*kubernetes.Clientset, client.Client) {
+	config, err := clientcmd.BuildConfigFromFlags("", filepath.Join(homedir.HomeDir(), ".kube", "config"))
+	Expect(err).NotTo(HaveOccurred())
+	clientSet, err := kubernetes.NewForConfig(config)
+	Expect(err).NotTo(HaveOccurred())
+
+	scheme := runtime.NewScheme()
```

Comment:
Not sure if this applies, but we do have this:
https://github.com/weaveworks/weave-gitops/blob/ee12fdf5957a9070994295e70ec5c2320cc3106f/pkg/kube/kubehttp.go#L35

Reply:
very nice! using this :smile: